drift/lib/src/runtime/manager/annotate.dart

```diff
@@ -0,0 +1,92 @@
+part of 'manager.dart';
+
+/// A class that contains the information needed to create an annotation
+sealed class BaseAnnotation<SqlType extends Object, $Table extends Table> {
+  /// The expression/column which will be added to the query
+  Expression<SqlType> get _expression;
+
+  /// The join builders that are needed to read the expression
+  final Set<JoinBuilder> _joinBuilders;
+  BaseAnnotation(this._joinBuilders);
+
+  /// Create an order by clause for this annotation
+  ColumnOrderings get order {
+    return ColumnOrderings(_expression);
+  }
+}
+
+/// A class that contains the information needed to create an annotation
+class Annotation<SqlType extends Object, $Table extends Table>
+    extends BaseAnnotation<SqlType, $Table> {
+  @override
+  final Expression<SqlType> _expression;
+
+  /// Create a filter for this annotation
+  ColumnFilters<SqlType> get filter {
+    return ColumnFilters(_expression);
+  }
+
+  /// Read the result of the annotation from the [BaseReferences] object
```

Comment:
I think this comment would benefit from a tiny snippet adding an annotation to a query and then reading it, for example. I think you can add the snippet we already have for the `withAnnotations` method.

Reply:
Done